sphinxcontrib/katex.py

```diff
@@ -138,31 +138,42 @@ def katex_autorenderer_content(app):
           renderMathInElement(document.body, katex_options);
         });
         ''')
-    katex_inline = ['\\' + d if d[0] == '\\' else d for d in app.config.katex_inline]
-    katex_display = ['\\' + d if d[0] == '\\' else d for d in app.config.katex_display]
-    katex_delimiters = {
-        'katex_inline': katex_inline,
-        'katex_display': katex_display
-    }
-    # Set chosen delimiters for the auto-rendering options of KaTeX
-    delimiters = r'''delimiters: [
-        {{ left: "{katex_inline[0]}", right: "{katex_inline[1]}", display: false }},
-        {{ left: "{katex_display[0]}", right: "{katex_display[1]}", display: true }}
-        ]'''.format(**katex_delimiters)
     prefix = 'katex_options = {'
     suffix = '}'
     options = katex_rendering_options(app)
+    delimiters = katex_rendering_delimiters(app)
     return '\n'.join([prefix, options, delimiters, suffix, content])
 
 
+def katex_rendering_delimiters(app):
+    """Delimiters for rendering KaTeX math.
+
+    If no delimiters are specified in katex_options, add the
+    katex_inline and katex_display delimiters. See also
+    https://khan.github.io/KaTeX/docs/autorender.html
+    """
+    # Return if we have user defined rendering delimiters
+    if 'delimiters' in app.config.katex_options:
+        return ''
+    katex_inline = [d.replace('\\', '\\\\') for d in app.config.katex_inline]
+    katex_display = [d.replace('\\', '\\\\') for d in app.config.katex_display]
+    katex_delimiters = {'inline': katex_inline, 'display': katex_display}
+    # Set chosen delimiters for the auto-rendering options of KaTeX
+    delimiters = r'''delimiters: [
+        {{ left: "{inline[0]}", right: "{inline[1]}", display: false }},
+        {{ left: "{display[0]}", right: "{display[1]}", display: true }}
+        ]'''.format(**katex_delimiters)
+    return delimiters
+
+
 def katex_rendering_options(app):
     """Strip katex_options from enclosing {} and append ,"""
     options = trim(app.config.katex_options)
     # Remove surrounding {}
     if options.startswith('{') and options.endswith('}'):
         options = trim(options[1:-1])
-    # Ensure list of options ends with ','
-    if not options.endswith(','):
+    # Ensure options is empty or ends with ','
```

Comment:
I think this should say "is not empty and ends with"

Reply:
`options` is allowed to be empty, or it can contain something, then it has to end with `,`. This is what the comment should state. I will try to improve it.